File: Swift_Backup/CORE/AreaLayout.swift
import SwiftUI

/// **AreaLayout**: Recursive description of the split screen layout grid.
/// Areas contain a unique UUID and a component type string showing what content is mounted.
public indirect enum AreaLayout: Codable, Hashable {
    case area(id: UUID, componentType: String)
    case split(direction: SplitDirection, ratio: CGFloat, first: AreaLayout, second: AreaLayout)
}

public enum SplitDirection: String, Codable {
    case horizontal
    case vertical
}

File: CORE/AreaLayout.ts
export type SplitDirection = 'horizontal' | 'vertical';

export interface LayoutArea {
  type: 'area';
  id: string;
  componentType: string;
}

export interface LayoutSplit {
  type: 'split';
  direction: SplitDirection;
  ratio: number;
  first: AreaLayout;
  second: AreaLayout;
}

export type AreaLayout = LayoutArea | LayoutSplit;
